### PR TITLE
Expand linting

### PR DESCRIPTION
**Pull request by kdzwinel**
- Extend linting to all JS  (excluding external code)
- Enable 'prefer-const' and 'eol-last'  in linter rules.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -248,7 +248,7 @@ module.exports = {
         "padded-blocks": "off",
         "padding-line-between-statements": "error",
         "prefer-arrow-callback": "error",
-        "prefer-const": "off",
+        "prefer-const": "error",
```
Comment (kdzwinel) [inline .eslintrc.js:251]: we should use 'const' for variables that we don't intend to override - this setting ensures that (https://eslint.org/docs/rules/prefer-const)

File: src/entities/shortNames.js [hunk 2]
```diff
@@ -33,30 +33,29 @@ function getDisplayName(name) {
...
 
             if (matchName.match(spacedRegex)) {
                 shortName = matchName.replace(spacedRegex, '')
                 break
             }
         } else if (uname.match(prefix)) {
-            let matchingPrefix = uname.match(prefix)[0]
-            let matchingSpacedPrefix = matchingPrefix.split('').join('\\W{0,1}')
-            let spacedRegex = new RegExp('^' + matchingSpacedPrefix + '\\W{0,1}\\s', 'i')
-            let matchName = name.replace(/^\./, '')
```
Comment (kdzwinel) [inline src/entities/shortNames.js:59]: This wasn't used - not sure if bug or not? cc @dharb

File: src/entities/update-entities.js [hunk 3]
```diff
@@ -16,17 +16,18 @@ const shortNames = require('./shortNames.js')
 const whoisRegexArray = whoisIgnoreList.map(regex => new RegExp(regex, 'gi'))
 const SSLRegexArray = SSLIgnoreList.map(regex => new RegExp(regex, 'gi'))
 const currentDate = new Date(Date.now()).toISOString()
+const WHOIS_XML_TOKEN = ''// credentials
```
Comment (kdzwinel) [inline src/entities/update-entities.js:19]: This wasn't defined - I'm guessing you are adding this line manually @dharb? Is this OK if I define it here like this?

File: src/entities/update-entities.js [hunk 4]
```diff
@@ -62,8 +63,8 @@ const filterSSLData = response => {
     // Depending on the length of the certificate chain, there may be multiple owner fields. We want the last one, which is most likely
     // to be the name of the actual company.
     // TODO check whois data for alt subjects on cert, add to entity if data matches
-    let sslNames = response.match(/O = .*,/g) || []
-    let sslAltNames = response.match(/DNS:.*/g) || []
+    const sslNames = response.match(/O = .*,/g) || []
+    // const sslAltNames = response.match(/DNS:.*/g) || []
```
Comment (kdzwinel) [inline src/entities/update-entities.js:67]: @dharb  also not used - not sure if a bug or just code that was left out

**Review by kdzwinel — COMMENTED** [review]
@Charlie-belmer @dharb @jdorweiler can y'all take a quick look (some of your code was modified) ? Thanks 🙌

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1 @@
+src/performance/helpers/lh/
```
Comment (kdzwinel) [inline .eslintignore:1]: Ignore Lighthouse code when linting.

File: .eslintrc.js [hunk 6]
```diff
@@ -69,7 +69,7 @@ module.exports = {
                 "allowKeywords": true
             }
         ],
-        "eol-last": "off",
+        "eol-last": "error",
```
Comment (kdzwinel) [inline .eslintrc.js:72]: github complains about lack of trailing EOLs, so lets make github happy - https://eslint.org/docs/rules/eol-last

File: .eslintignore (hunk 5 above)
Comment (kdzwinel) [inline .eslintignore:1]: Ignore Lighthouse code when linting.

File: src/entities/shortNames.js (hunk 2 above)
Comment (dharb) [reply src/entities/shortNames.js:59]: Ah yeah looks like a remnant of old code that I revised.

File: src/entities/update-entities.js (hunk 3 above)
Comment (dharb) [reply src/entities/update-entities.js:19]: Yeah we use our own access token for the api. Works for me 👍

File: src/entities/update-entities.js (hunk 4 above)
Comment (dharb) [reply src/entities/update-entities.js:67]: Yeah remnants of more code that I removed. When I originally built the entity list I pulled in domains listed under SSL alt names. It was great for building out breadth for each entity, but it unfortunately produced a lot of false positives that I had to filter out (due to shared SSL certs).

I decided to remove it from the update script and currently use a different script that requires a manual check before anything is added to our data. I'd like to revisit at some point and get that in open source as well, but just haven't had a chance.

**Review by dharb — APPROVED** [review]
Looks good to me, thanks for doing this @kdzwinel.

**Review by jdorweiler — APPROVED** [review]

**Pull request merged**